### PR TITLE
Add support for `if_exists/if_not_exists` on `remove_foreign_key/add_foreign_key`

### PR DESCRIPTION
**Pull request by robertomiranda**
### Summary

Adding support for `if_exists/if_not_exists` on `remove_foreign_key/add_foreign_key`. Applications can set their migrations to ignore exceptions raised when adding a foreign key that already exists or when removing a foreign key that does not exist.

Similar behaviour has been implemented on: 

- add_column/remove_column https://github.com/rails/rails/pull/38352
- add_index https://github.com/rails/rails/pull/38555
- remove_index https://github.com/rails/rails/pull/38967

### CODE REVIEW

File: activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb
Comment (ghiculescu) [inline activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb:1073]: The comments above should be updated, that’ll become the docs for this.

**Comment by ghiculescu** [comment]
Could you squash and add a changelog?

**Comment by robertomiranda** [comment]
thanks, @ghiculescu 🙇 , Suggestions applied!

> Could you squash and add a changelog?

Out of curiosity, is there any reason why [Squash and merge](https://docs.github.com/en/github/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-pull-request-commits) feature is not used on this repository?

**Comment by ghiculescu** [comment]
Not 100% sure, I’m guessing because it moves responsibility for the final commit message from the author to the maintainer, which creates more work for maintainers.

File: activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb
Comment (guilleiguaran) [inline activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb:1075]: This example should go above with the other examples

**Review by guilleiguaran — CHANGES_REQUESTED** [review]
One small comment before merging this

**Review by guilleiguaran — APPROVED** [review]

**Pull request merged**
